terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+-rw-r--r--  1 danielvilla  staff  22365 Feb 18 20:48 linux.png
+```
+Awesome, but that's kinda confusing, what are those numbers next to the staff word? Most commands that deal with storage properties share a commond flag **-h** that means "human readable"
+
+```
+danielvilla@Daniels-MacBook-Air:~/docker/linux_tweet_app$ ls -lh
+total 88
+-rw-r--r--  1 danielvilla  staff   147B Feb 18 20:48 Dockerfile
+-rw-r--r--  1 danielvilla  staff   284B Feb 18 20:48 README.md
+-rw-r--r--  1 danielvilla  staff   1.6K Feb 18 20:48 index-new.html
+-rw-r--r--  1 danielvilla  staff   1.5K Feb 18 20:48 index-original.html
+-rw-r--r--  1 danielvilla  staff   1.6K Feb 18 20:48 index.html
+-rw-r--r--  1 danielvilla  staff    22K Feb 18 20:48 linux.png
+```
+Now we can see a more understandable output regarding the file size, and as you might guess Feb 18 is the date when the file was created. Don't worry about the other stuff for now
+
+#### Printing and env variables
+
+Now you might be thinking where does this all come from? Where are we? What's the space the terminal is printing in? Where does the prompt come from? 
+
+Well first let's learn about the **printenv** and **set** commands. As every program the terminal has it's own defined variables in it's *environment*
+
+printenv and set (without any arguments) print the defined variables on the session
+
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ printenv
+XDG_SESSION_ID=4
+TERM=xterm-256color
+SHELL=/bin/bash
+SSH_CLIENT=10.0.2.2 50356 22
+SSH_TTY=/dev/pts/0
+USER=vagrant
+```
+and
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ set
+BASH=/bin/bash
+BASHOPTS=checkwinsize:cmdhist:complete_fullquote:expand_aliases:extglob:extquote:force_fignore:histappend:interactive_comments:login_shell:progcomp:promptvars:sourcepath
+BASH_ALIASES=()
+BASH_ARGC=()
+BASH_ARGV=()
+```
+but there's particulary one variable that we want to see and that is **PS1** that controls the look of the shell prompt.
+
+But how do we see what the PS1 variable contains? The command we are looking for to print strings and variables is **echo**, let's try it with always classy hello world
+
+```
+danielvilla@Daniels-MacBook-Air:~$ echo "Hello world"
+Hello world
+```
+now lets see what the PS1 variable has
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ echo $PS1
+\[\e]0;\u@\h: \w\a\]${debian_chroot:+($debian_chroot)}\u@\h:\w\$
+```
+Notice that we had to use **$** sign before the PS1 variable name, it's not used when we inizialiate it, but it is when call it.
+
+Now, yeah, that variable is printed every time you hit enter, that's how we see the prompt, so if we change the value of the $PS1 variable we will see another prompt
+
+```
+export PS1="new prompt"
+```
+Yup, that's your new prompt, to go back just add again the previous value
+
+
+
+#### Other casual commands
+
+These are another handful of useful commands, to give you more vision of your environment while using the terminal, when everything seems dark.
```

Comment:
These are another handful of useful commands ~, to~ `that` give you more vision of your environment while using the terminal, when everything seems dark.